compiler/parser/src/parser/file.rs

```diff
@@ -349,7 +349,11 @@ impl<N: Network> ParserContext<'_, N> {
         let name = self.expect_identifier()?;
 
         // Parse parameters.
-        let (inputs, ..) = self.parse_paren_comma_list(|p| p.parse_input().map(Some))?;
+        let (inputs, _, span) = self.parse_paren_comma_list(|p| p.parse_input().map(Some))?;
+
+        if inputs.is_empty() && matches!(variant, Variant::Function) {
```

Comment:
While it's fine to make this a parser error, it just occurred to me that it would also be appropriate to make this a compiler error, which would also make handling the impacted test cases easier. Also by making it a compiler error, it leaves the grammar unchanged. However, I'm on board for either.